src/server/namespaces.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/node_hash_map.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "server/blocking_controller.h"
+#include "server/db_slice.h"
+#include "server/tx_base.h"
+#include "util/fibers/synchronization.h"
+
+namespace dfly {
+
+class Namespace {
```

Comment:
maybe add class description, including a threading/ownership model?
can also be a reference to .md file if you write it there.